src/hotspot/share/include/jvm.h

```diff
@@ -276,7 +276,7 @@ JNIEXPORT void JNICALL
 JVM_Yield(JNIEnv *env, jclass threadClass);
 
 JNIEXPORT void JNICALL
-JVM_Sleep(JNIEnv *env, jclass threadClass, jlong millis);
+JVM_Sleep(JNIEnv *env, jclass threadClass, jlong millis, jint nanos);
```

Comment:
I wonder if it would be simpler to just provide a single value, in nanoseconds, to the VM. That's enough for a sleep of 292 years. Windows would still need to convert to milliseconds of course but it overall would avoid sending two values down to the park code.

Reply:
Right, that might be better. New revision should address that.